src/main/java/com/powsybl/openloadflow/ac/outerloop/AreaInterchangeControlOuterloop.java

```diff
@@ -0,0 +1,219 @@
+                .map(entry -> String.format(Locale.US, "%s: %.2f MW", entry.getKey().getId(), entry.getValue() * PerUnit.SB))
+                .collect(Collectors.joining(", "));
+    }
+
+    private double getInterchangeMismatch(LfArea area) {
+        return area.getInterchangeTarget() - area.getInterchange();
+    }
+
+    private double getInterchangeMismatchWithSlack(LfArea area, double slackBusActivePowerMismatch) {
+        return area.getInterchange() - area.getInterchangeTarget() + getSlackInjection(area, slackBusActivePowerMismatch);
+    }
+
+    private double getSlackInjection(LfArea area, double slackBusActivePowerMismatch) {
+        return areaSlackDistributionParticipationFactor.getOrDefault(area, 0.0) * slackBusActivePowerMismatch;
+    }
+
+    private static void reportAndLogSuccess(ReportNode reportNode, Map<LfArea, Double> areaInterchangeMismatches, int areasCount, int iterationCount) {
+        String initialMismatches = mismatchesToString(areaInterchangeMismatches);
+        Reports.reportAreaMismatchDistributionSuccess(reportNode, initialMismatches, areasCount, iterationCount);
+        LOGGER.info("Area Interchange mismatches [{}] distributed in {} distribution iteration(s)",
+                initialMismatches, iterationCount);
+    }
+
+    private void allocateSlackDistributionParticipationFactors(LfNetwork lfNetwork) {
+        List<LfBus> slackBuses = lfNetwork.getSlackBuses();
+        int totalSlackBusCount = slackBuses.size();
+        for (LfBus slackBus : slackBuses) {
+            Optional<LfArea> areaOpt = slackBus.getArea();
+            if (areaOpt.isPresent()) {
+                areaSlackDistributionParticipationFactor.put(areaOpt.get(), areaSlackDistributionParticipationFactor.getOrDefault(areaOpt.get(), 0.0) + 1.0 / totalSlackBusCount);
+            } else {
+//                When a bus is connected to one or multiple Areas but the flow through the bus is not considered for those areas' interchange power flow,
+//                its slack injection should be considered for the slack of some Areas that it is connected to.
+                Set<LfBranch> connectedBranches = new HashSet<>(slackBus.getBranches());
+                Set<LfArea> connectedAreas = connectedBranches.stream()
+                        .flatMap(branch -> Stream.of(branch.getBus1(), branch.getBus2()))
+                        .filter(Objects::nonNull)
+                        .map(LfBus::getArea)
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(Collectors.toSet());
+
+                Set<LfArea> areasSharingSlack = connectedAreas.stream()
+                                .filter(area -> area.getBoundaries().stream().noneMatch(boundary -> connectedBranches.contains(boundary.getBranch())))
+                                .collect(Collectors.toSet());
+                if (!areasSharingSlack.isEmpty()) {
```

Comment:
Add comments to explain what it tested/done here :
// if the slack bus is on a boundary point considered for net position, 
// it will resolve naturally because deviations caused by slack are already present on tie line flows
// no need to include in any area net position calculation

Reply:
added in dae8119